### PR TITLE
fix test expectations

### LINKED ISSUE #6312: test_get_func_args() expectation changes in new Python point releases
See https://github.com/scrapy/itemloaders/issues/76

### PR DESCRIPTION
**Pull request by kokobhara**
Fixes #6312

### CODE REVIEW

File: tests/test_utils_python.py [hunk 1]
```diff
@@ -238,13 +237,11 @@ def __call__(self, a, b, c):
...
-            # doesn't work on CPython: https://bugs.python.org/issue42785
-            self.assertEqual(get_func_args(operator.itemgetter(2)), [])
-        elif platform.python_implementation() == "PyPy":
-            self.assertEqual(
-                get_func_args(operator.itemgetter(2), stripself=True), ["obj"]
-            )
+        # Didn't used to work on CPython: https://bugs.python.org/issue42785
+        self.assertIn(
+            get_func_args(operator.itemgetter(2), stripself=True),
+            [[], ["obj"], ["args", "kwargs"]],
```
Comment (Gallaecio) [inline tests/test_utils_python.py:243]: Is `["args", "kwargs"]` what CPython returns now? If so, it might be best to keep the old if/else for PyPy, and assert CPython for `[[], ["args", "kwargs"]]` instead.

Comment (kokobhara) [reply tests/test_utils_python.py:243]: Alright, I'll update it.

**Review by Gallaecio — APPROVED** [review]

File: tests/test_utils_python.py [hunk 2]
```diff
@@ -239,8 +239,11 @@ def __call__(self, a, b, c):
         self.assertEqual(get_func_args(" ".join, stripself=True), ["iterable"])
 
         if platform.python_implementation() == "CPython":
-            # doesn't work on CPython: https://bugs.python.org/issue42785
-            self.assertEqual(get_func_args(operator.itemgetter(2)), [])
+            # Didn't used to work on CPython: https://bugs.python.org/issue42785
```
Comment (wRAR) [inline tests/test_utils_python.py:242]: I'd rephrase it: https://english.stackexchange.com/questions/8816/whats-the-negation-of-i-used-to-be-surely-not-i-didnt-used-to-be

Comment (kokobhara) [reply tests/test_utils_python.py]: Updated

File: tests/test_utils_python.py [hunk 3]
```diff
@@ -239,8 +239,11 @@ def __call__(self, a, b, c):
         self.assertEqual(get_func_args(" ".join, stripself=True), ["iterable"])
 
         if platform.python_implementation() == "CPython":
-            # doesn't work on CPython: https://bugs.python.org/issue42785
-            self.assertEqual(get_func_args(operator.itemgetter(2)), [])
+            # This never worked on CPython: https://bugs.python.org/issue42785
```
Comment (wRAR) [inline tests/test_utils_python.py:242]: Well it didn't work but now works :) How about "This didn't work on older versions of CPython"?

Also I would update the issue URL to https://github.com/python/cpython/issues/86951

**Review by wRAR — APPROVED** [review]

**Comment by wRAR** [comment]
Thanks!

**Comment by kokobhara** [comment]
Should I make a similar PR for [scrapy/itemloaders#76](https://github.com/scrapy/itemloaders/issues/76)?

**Comment by Gallaecio** [comment]
> Should I make a similar PR for [scrapy/itemloaders#76](https://github.com/scrapy/itemloaders/issues/76)?

That would be awesome!
